moveit_ros/planning/planning_scene_monitor/src/current_state_monitor.cpp

```diff
@@ -142,10 +143,14 @@ void planning_scene_monitor::CurrentStateMonitor::startStateMonitor(const std::s
           joint_states_topic, 25, std::bind(&CurrentStateMonitor::jointStateCallback, this, std::placeholders::_1));
     if (tf_buffer_ && !robot_model_->getMultiDOFJointModels().empty())
     {
-      // TODO (anasarrak): replace this for the appropiate function, there is no similar
-      // function in ros2/geometry2.
-      // tf_connection_.reset(new TFConnection(
-      //     tf_buffer_->_addTransformsChangedListener(std::bind(&CurrentStateMonitor::tfCallback, this))));
+      using callback_t = std::function<void(const tf2_msgs::msg::TFMessage::ConstSharedPtr)>;
+      tf_buffer_->setUsingDedicatedThread(true);
+      transform_subscriber_ = node_->create_subscription<tf2_msgs::msg::TFMessage>(
+          "/tf", tf2_ros::DynamicListenerQoS(25),
+          callback_t(std::bind(&CurrentStateMonitor::transfromCallback, this, std::placeholders::_1, false)));
+      static_transfrom_subscriber_ = node_->create_subscription<tf2_msgs::msg::TFMessage>(
+          "/tf_static", tf2_ros::StaticListenerQoS(25),
+          callback_t(std::bind(&CurrentStateMonitor::transfromCallback, this, std::placeholders::_1, true)));
```

Comment:
Would it make sense to wrap all this in a new class that takes `std::shared_ptr<rclcpp::Node>` and a `bind` to a callback like `updateMultiDofJoints`? That seems like a useful bit of capability to have as a portable building block. As written it seems like things are a bit scattered between different files.

(I think @henningkayser had suggested something like this previously).

Reply:
I tried implementing the class I described earlier and I think I've changed my mind about it being especially useful.

The `transformCallback` needs to have some built-in logic specifically designed around handling static and non-static TF callbacks from a pair of subscribers, so it's not as straightforward to allow a user to pass in an arbitrary callback and have it work in an equivalent way to the current method.